File: ProblemSolve/121.cpp
/***********************************
 * Best Time to Buy and Sell Stock *
 ***********************************/

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int len = prices.size();
        int profit = 0;
        int chose = 10001;
        
        //최저점을 갱신하면서 최저점에 따른 최고수익을 계산
        for(int i=0; i<len; i++){
            chose = min(chose, prices[i]);
            profit = max(profit, prices[i] - chose);
        }
        
        return profit;        
    }
};

File: ProblemSolve/121.cs
/***********************************
 * Best Time to Buy and Sell Stock *
 ***********************************/

public class Solution {
    public int MaxProfit(int[] prices) {
        int ans = 0;
        int buyPrice = 10001;
        
        for(int i=0; i<prices.Length; ++i){            
            buyPrice = Math.Min(prices[i], buyPrice);
            ans = Math.Max(ans, prices[i] - buyPrice);
        }
        
        return ans;
    }
}